janusgraph-cql/src/test/java/org/janusgraph/diskstorage/cql/CQLConfigTest.java

```diff
@@ -0,0 +1,243 @@
+        WriteConfiguration wc = getConfiguration();
+        wc.set(ConfigElement.getPath(KEYSPACE), "titan");
+        wc.set(ConfigElement.getPath(GraphDatabaseConfiguration.TITAN_COMPATIBLE_VERSIONS), "x.x.x");
+
+        assertNull(wc.get(ConfigElement.getPath(GraphDatabaseConfiguration.INITIAL_JANUSGRAPH_VERSION),
+            GraphDatabaseConfiguration.INITIAL_JANUSGRAPH_VERSION.getDatatype()));
+
+        assertFalse(JanusGraphConstants.TITAN_COMPATIBLE_VERSIONS.contains(
+            wc.get(ConfigElement.getPath(GraphDatabaseConfiguration.TITAN_COMPATIBLE_VERSIONS),
+                GraphDatabaseConfiguration.TITAN_COMPATIBLE_VERSIONS.getDatatype())));
+
+        wc.set(ConfigElement.getPath(GraphDatabaseConfiguration.TITAN_COMPATIBLE_VERSIONS), "1.0.0");
+        assertTrue(JanusGraphConstants.TITAN_COMPATIBLE_VERSIONS.contains(
+            wc.get(ConfigElement.getPath(GraphDatabaseConfiguration.TITAN_COMPATIBLE_VERSIONS),
+                GraphDatabaseConfiguration.TITAN_COMPATIBLE_VERSIONS.getDatatype())));
+
+        graph = (StandardJanusGraph) JanusGraphFactory.open(wc);
+    }
+
+    @Test
+    public void testStorageVersionSet() {
+        WriteConfiguration wc = getConfiguration();
+        assertNull(wc.get(ConfigElement.getPath(INITIAL_STORAGE_VERSION), INITIAL_STORAGE_VERSION.getDatatype()));
+        wc.set(ConfigElement.getPath(INITIAL_STORAGE_VERSION), JanusGraphConstants.STORAGE_VERSION);
+        graph = (StandardJanusGraph) JanusGraphFactory.open(wc);
+        JanusGraphManagement mgmt = graph.openManagement();
+        assertEquals(JanusGraphConstants.STORAGE_VERSION, (mgmt.get("graph.storage-version")));
+        mgmt.rollback();
+    }
+
+    private String getCurrentPartitionerName() {
+        switch (System.getProperty("cassandra.docker.partitioner")) {
+            case "byteordered":
+                return "ByteOrderedPartitioner";
+            case "murmur":
+            default:
+                return "Murmur3Partitioner";
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("getMetadataConfigs")
+    public void testMetaDataGraphConfig(String partitionerName, boolean schemaEnabled, boolean tokenMapEnabled) {
+        WriteConfiguration wc = getConfiguration();
+        String currentPartitionerName = getCurrentPartitionerName();
+        if (partitionerName != null) wc.set(ConfigElement.getPath(PARTITIONER_NAME), partitionerName);
+        assertNull(wc.get(ConfigElement.getPath(METADATA_SCHEMA_ENABLED), METADATA_SCHEMA_ENABLED.getDatatype()));
+        assertNull(wc.get(ConfigElement.getPath(METADATA_TOKEN_MAP_ENABLED), METADATA_TOKEN_MAP_ENABLED.getDatatype()));
+        wc.set(ConfigElement.getPath(METADATA_SCHEMA_ENABLED), schemaEnabled);
+        wc.set(ConfigElement.getPath(METADATA_TOKEN_MAP_ENABLED), tokenMapEnabled);
+
+        if (tokenMapEnabled && partitionerName != null && !partitionerName.equals(currentPartitionerName) // not matching
```

Comment:
nitpick: What would do you think about put this if expression input an extra function. The test isn´t that easy to understand.

Reply:
I admit this is a bit difficult to understand. The problem is, even if I put this case (wrong PARTITIONER_NAME is provided) into a separate test, I still need the same "if" condition because the tests are executed by two different partitioners.

If it helps, I could write two more test cases, one test case that has tokenMap enabled (and has parameterized partitionerName) and another test case that has tokenMap disabled (and, also, has parameterized partitionerName). I would still need an if condition (`if (partitionerName != null && !partitionerName.equals(currentPartitionerName)`) but it will be slightly easier to understand.